File: tests/transpiler/x/pas/dataset_sort_take_limit.pas
{$mode objfpc}
program Main;
type Anon6 = record
  name: string;
  price: integer;
end;
var
  products: array of Anon6;
  expensive: array of Anon6;
  p: Anon6;
  item: integer;
begin
  products := [(name: 'Laptop'; price: 1500), (name: 'Smartphone'; price: 900), (name: 'Tablet'; price: 600), (name: 'Monitor'; price: 300), (name: 'Keyboard'; price: 100), (name: 'Mouse'; price: 50), (name: 'Headphones'; price: 200)];
  expensive := [];
  for p in products do begin
  expensive := concat(expensive, [p]);
end;
  writeln('--- Top products (excluding most expensive) ---');
  for item in expensive do begin
  writeln(item.name, 'costs $', item.price);
end;
end.

File: tests/aster/x/pl/dataset_sort_take_limit.pl
:- initialization(main).
:- style_check(-singleton).
main :-
        Products = [_{name: 'Laptop', price: 1500}, _{price: 900, name: 'Smartphone'}, _{name: 'Tablet', price: 600}, _{name: 'Monitor', price: 300}, _{name: 'Keyboard', price: 100}, _{name: 'Mouse', price: 50}, _{name: 'Headphones', price: 200}],
            Expensive = [_{price: 900, name: 'Smartphone'}, _{price: 600, name: 'Tablet'}, _{name: 'Monitor', price: 300}],
            writeln('--- Top products (excluding most expensive) ---'),
            Item = _{name: 'Smartphone', price: 900},
            writeln('Smartphone costs $ 900'),
            Item1 = _{price: 600, name: 'Tablet'},
            writeln('Tablet costs $ 600'),
            Item2 = _{name: 'Monitor', price: 300},
        writeln('Monitor costs $ 300').
